tests/Unit/Adapter/Hook/HookDispatcherTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Tests\Unit\Adapter\Hook;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Hook\HookDispatcher;
+use PrestaShopBundle\Service\Hook\HookEvent;
+
+class HookDispatcherTest extends TestCase
+{
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|HookEvent
+     */
+    private $hookEventMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|HookDispatcher
+     */
+    private $hookDispatcherMock;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->hookEventMock = $this
+            ->getMockBuilder(HookEvent::class)
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $this->hookDispatcherMock = $this
+            ->getMockBuilder(HookDispatcher::class)
+            ->setMethods(['getListeners', 'doDispatch'])
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+    }
+
+    /**
+     * @dataProvider getHookEventNames
+     *
+     * @param string $eventName
+     */
+    public function testItCallsHookDispatcherWithCaseInsensitiveEventNames($eventName)
+    {
+        $lowerCasedEventName = strtolower($eventName);
+
+        $this->hookDispatcherMock
+            ->method('getListeners')
+            ->with($lowerCasedEventName)
```

Comment:
@PierreRambaud  just to be clear - we were talking about adding `->with` in this place , right?

Reply:
No, you are not testing this function properly: `$this->doDispatch($listeners, $eventName, $event);`

You check only if  `$this->doDispatch();` is called.
Must be something like:
```php
$lowerCasedEventName = strtolower($eventName);
$this->hookDispatcherMock
    ->method('getListeners')
    ->with($lowerCasedEventName)
    ->willReturn([]); // getListeners returns an array, cf syfmony doc

$this->hookDispatcherMock
    ->expects($this->once())
    ->method('doDispatch')
    ->with([], $lowerCasedEventName, $this->hookEvent);
$this->hookDispatcherMock->dispatch($eventName, $this->hookEventMock);
```

Plus `getListeners` returns an array.

Reply:
yes it makes sence - thank you!